languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java

```diff
@@ -0,0 +1,97 @@
+/* LanguageTool, a natural language style checker
+ * Copyright (C) 2005 Daniel Naber (http://www.danielnaber.de)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+package org.languagetool.rules.ga;
+
+import org.languagetool.AnalyzedSentence;
+import org.languagetool.AnalyzedToken;
+import org.languagetool.AnalyzedTokenReadings;
+import org.languagetool.rules.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ResourceBundle;
+
+import static org.languagetool.rules.ga.DhaNoBeirtData.getNumberReplacements;
+import static org.languagetool.rules.ga.DhaNoBeirtData.getDaoine;
+
+public class DhaNoBeirtRule extends Rule {
+  public DhaNoBeirtRule(ResourceBundle messages) {
+    super.setCategory(Categories.MISC.getCategory(messages));
+    setLocQualityIssueType(ITSIssueType.Misspelling);
+    addExamplePair(Example.wrong("Tá <marker>dhá</marker> dheartháireacha agam."),
+      Example.fixed("Tá <marker>beirt</marker> deartháireacha agam."));
+  }
+
+  @Override
+  public String getId() {
+    return "GA_DHA_NO_BEIRT";
+  }
+
+  @Override
+  public String getDescription() {
+    return "'dhá' nó 'beirt'";
+  }
+
+  @Override
+  public RuleMatch[] match(AnalyzedSentence sentence) {
+    List<RuleMatch> ruleMatches = new ArrayList<>();
+    AnalyzedTokenReadings[] tokens = sentence.getTokensWithoutWhitespace();
+    int markEnd = 1;
+    String replacement = null;
+    String msg = null;
+    for (int i = 1; i < tokens.length; i++) {  // ignoring token 0, i.e., SENT_START
+      if (isNumber(tokens[i]) && (i < tokens.length - 1 && isPerson(tokens[i + 1]))) {
+        markEnd = i + 1;
+        if ("dhá".equalsIgnoreCase(tokens[i].getToken())) {
+          for (int j = i + 1; j < tokens.length; j++) {
+            if ("déag".equalsIgnoreCase(tokens[j].getToken())) {
+              markEnd = j;
+              replacement = "dháréag";
+              msg = "";
```

Comment:
Probably needs the intended string here. Maybe "Ní mór duit an uimhir uimhriúil phearsanta a úsáid a thagraíonn do dhaoine.", if this machine translation is ok.

Reply:
I spent years working on MT, and in general, it's not ok for Irish (too little training data available). That MT output suggests a style rule, perhaps: 'uimhir uimhriúil' is 'numerical number'

Reply:
I have seen some of your work in Apertium. Really appreciated.
In this case, it might have been my fault. My input was "personal numeral" which I am not even sure is correct in English. 
I have added a MTed MessageBundle to Irish, that will work well here as is. If you add it to #915 it will be probably overwritten by Transifex later on. However, if you find any egregious error in your testing, please tell me and I will delete the file.